js/songs/edit.controller.js

```diff
@@ -0,0 +1,24 @@
+"use strict";
+
+(function(){
+  angular
+    .module( "songs" )
+    .controller( "songEditController", [
+      "songFactory",
+      "$stateParams",
+      "$state",
+      songEditControllerFunction
+    ]);
+
+  function songEditControllerFunction( songFactory, $stateParams, $state ){
+    this.song = songFactory.get({id: $stateParams.id});
+    this.update = function(){
+      this.song.$update({id: $stateParams.id});
+      $state.go("songIndex", {}, { reload: true }); //these sometimes work properly and sometimes don't work properly, don't really know why.
```

Comment:
Hmm...it's interesting that it's not working consistently. It leads me to think that it's trying to load the new state before the update has finished, so I would try adding in a promise: 
```
this.song.$update({id: $stateParams.id}).then(function() {
$state.go("songIndex", {}, { reload: true })
});

```